### ISSUE TITLE
Google Music: playback stops at the end of a song and playlist stalls

### ISSUE THREAD

**Issue by tizonia** [tizonia/tizonia-openmax-il#489]
### Description

Sometimes (e.g 1 in 5-10) songs, the playback stops at the end of a song, and the next song in the playlist is never reproduced. The player is alive since pressing 'n' or 'p' will make it continue playing songs.

### Steps to Reproduce

1. Use some Google Music search option.
2. Leave it running alone for a few songs.
3. At some point, playback stops at the end of a song, and the next song is not reprduced.

**Expected behavior:** 
No interruptions in the playlist progress (the next song in the playlist is always reproduced).

**Actual behaviour:** 
Sometimes the playback stops (e.g. 1 in 5-10 songs)

**Reproduces how often:**
Quite often. It has been seen with Raspbian and Ubuntu 18.04 amd64 bintray debs.

### Versions

Version 0.14.0

### Installation Method Used

Bintray debs

### Additional Information

N/A.

**Comment by ejiek** [comment]
Same behaviour for AUR build.

I guess it's the same problem that 0.12.0 suffers from (only personal experience):

When track is long enough Google changes tracks link, so when tizonia tries to download the missing part it fails. 0.12.0 used to start playing next track when the current fails. 0.14.0 just stoppes.

**Comment by tizonia** [comment]
Very good analysis. 

There is a fix (12a1a87) in the develop branch that seems to remove the stalling problem. I have not released it yet because of lack of time. Hopefully, 0.15 will be out in the next couple of weeks with this fix.

**Comment by juanrubio** [comment]
Just an update on this issue:

The initial problem was fixed with 12a1a87. This makes the initial player stalling problem go away (which used to occur after 5-10 songs). However, after some more testing, the player still stalls, but it just takes longer, around 20-30 songs, until another issue shows up.

This issue now appears to be in the lower layers, either libcurl, or GnuTLS:
```
libcurl : [SSL re-using session ID
libcurl : [gnutls_handshake() failed: An unexpected TLS packet was received.
libcurl : [stopped the pause stream!
libcurl : [Closing connection 28
```

I've currently updated Tizonia's logic to retry the libcurl download, but unfortunately, the gnutls_handshake error keeps showing up. So there is something that seems broken at that point in the underlaying software which renders the scenario unrecoverable.

I'm currently discussing this issue in the curl-library mailing list:
https://curl.haxx.se/mail/lib-2018-06/0035.html
https://curl.haxx.se/mail/lib-2018-06/0038.html
https://curl.haxx.se/mail/lib-2018-06/0045.html
